back-end/utils/top-layer/UserUtil.js

```diff
@@ -362,6 +362,173 @@ const getLengthUserTransactionsHistoryForRedisM5RU = (email, filters) => {
   });
 };
 
+/**
+ * Perform buy action for the function proceedTransaction
+ * @param {import(".prisma/client").User} user
+ * @param {number} totalPendingPrice
+ * @param {string} companyCode
+ * @returns
+ */
+ const buyShareEvent = async (
+  user,
+  totalPendingPrice,
+  companyCode,
+  quantity,
+  recentPrice
+) => {
+  try {
+    if (user.cash < totalPendingPrice) {
+      return new Error(
+        "The user does not have enough money to buy the pending stock"
+      );
+    }
+
+    user.cash -= totalPendingPrice;
```

Comment:
This line does not update user in database. Use prisma.user.update to update user cash.